doc/user/guide/evaluation.rst

```diff
@@ -1,2 +1,172 @@
+
+`Nushell <https://www.nushell.sh>`__ can be installed from ``Homebrew``.
+
+::
+
+   brew install nushell
+
+Save this file as ``sequant.nu`` in the same directory where the trace
+output file is present. We will call this directory 'working directory'.
+
+.. code:: nu
+
+
+   use std/assert
+
+   const pattern = {
+     trace: '^\s*(Term|Eval|Cache)\s+'
+     separator: '|'
+   }
+
+   def "read log" [file: string] {
+       grep -n -E -e $pattern.trace $file
+         | parse --regex '(?<line>\d+):\s+(?<trace>.*)'
+         | each {
+           let line = $in.line
+           let fields = $in.trace
+                        | split row $pattern.separator 
+                        | str trim
+           match $fields {
+             [$tag0 $tag1 ..$rest] => {
+                         {tag0 : $tag0
+                          tag1 : $tag1
+                          trace: $rest
+                          line : ($line | into int) } }
+             }
+           }
+   }
+
+   def "new-cols eval" [] {
+       {
+         time   : ($in.0 | into duration)
+         memory : ($in.1 | into filesize)
+         annot  : $in.2
+       }
+   }
+
+   def "new-cols cache" [] {
+     {
+       key    : $in.0
+       life   : $in.1
+       nalive : ($in.2 | into int)
+       memory : ($in.3 | into filesize )
+     }
+   }
+
+   def make-tag [tag: string, make_col: closure] {
+     let entry = $in | where tag0 == $tag
+     let new_cols = $entry.trace | each { do $make_col }
+     $entry | select tag0 tag1
+            | merge $new_cols
+            | merge ($entry | select line)
+   }
+
+   export def "trace cache" [file: string] {
+     read log $file | make-tag 'Cache' { new-cols cache }
+   }
+
+   export def "trace eval" [
+               file: string # The text file with sequant evaluation trace.
+               ] {
+     read log $file | make-tag 'Eval' { new-cols eval }
+   }
+
+   export def "trace term" [file: string] {
+     let lines = read log $file | where tag0 == 'Term'
+     let begs  = $lines | every 2
+     let ends  = $lines | skip 1 | every 2
+     $begs | zip $ends
+           | each { match $in { [$b, $e] => { assert ($b.trace.0 == $e.trace.0)
+                                              { tag0  : $b.tag0
+                                                begin : $b.line
+                                                end   : $e.line
+                                                expr  : $b.trace.0
+                                              } } } }
+   }
+
+This nu script exports the functions: ``trace eval``, ``trace cache``,
+and ``trace term``.
+
+If you are using a different shell (eg. bash, zsh, etc.), launch the
+nushell from the working directory. Make sure the executable ``nu``
+installed from Homebrew is on your path.
+
+.. code:: sh
+
+   nu
+
+Then import the functions from the script.
+
+.. code:: nu
+
+   use sequant.nu *
+
+Now we analyze the output in steps.
+
+.. code:: nu
+
+   let eval = trace eval out.txt
```

Comment:
this needs to become a complete example. Right now it's not clear what's in out.txt, how it was produced, etc.

Reply:
Added a sample `doc/user/guide/trace.txt` that is explored in the tutorial.